libs/ngx-nuts-and-bolts/testing-utils/README.md

```diff
@@ -105,3 +105,49 @@ describe('asyncError demo', () => {
 	});
 });
 ```
+
+# `asyncData`
+
+If and when returning mocked data from the testing service, users should avoid doing something like this:
+
+```ts
+public getData(data: TData): Observable<TData>{
+	return of(data);
+}
+```
+
+The example above creates an observable from the data but it [returns it synchronously](https://stackblitz.com/edit/rxjs-hx8yos?devtoolsheight=60). This isn't how services that make API calls behave in real world applications.
+The purpose of creating test doubles for services is to mock the behavior of real world services as closely as possible. With that in mind, it is a good idea to fake the async behavior of such services. This can can be achieved in multiple ways.
+This specific implementation uses `observeOn` operator with `asyncScheduler`, creating a new macrotask that places the observable in the event loop queue.
+
+## 1. Usage
+
+When returning the mock data in test doubles, simply wrap the data using `asyncData` function and the data will be returned asynchronously.
+
+```ts
+public getTestingData(data: TData): Observable<TData>{
+	return asyncData(data);
+}
+```
+
+# `MockStorage`
+
+When you need to interact with storage but still you are testing only a unit of the code, you can use `MockStorage` class for this purpose. Mocked version doesn't take into account stuff like size limits and other things taken care by `window.localstorage API` and just focuses on core functionality.
+
+## 1. Usage
+
+```ts
+let storage: Storage;
+
+beforeEach(async () => {
+	await TestBed.configureTestingModule({
+		providers: [
+			{
+				provide: Storage,
```

Comment:
Please update the providers here as well, not just in docs.

We can think of keeping docs just in one place so we do not have to c/p to keep changes in-sync in both places.

Reply:
Yeah, you have to keep this in mind always. I think we could forgo the readme files for secondary entry points and just reference docusaurus .md files since they have some additional metadata thats needed for docs generation.